mono/mini/genmdesc.py

```diff
@@ -146,53 +146,92 @@ def parse_input(infile, opcodes):
                 print ("Unknown specifier '{0}' for opcode '{0}'".format (spec))
                 exit (1)
 
-def gen_output(f, opcodes):
+def gen_output(f, opcodes):
     sorted_opcodes = []
     for op in opcodes.values ():
         sorted_opcodes.append (op)
     sorted_opcodes.sort (key=lambda op: op.num)
 
     f.write ("/* File automatically generated by genmdesc.py, don't change */\n\n")
 
-    # Write desc table
-
-    f.write ("const MonoInstSpec mono_{0} [] = {{\n".format (symbol_name))
-    f.write ("  {{")
-
+    # Write desc table
+    f.write ("const MonoInstSpec mono_{0} [] = {{\n".format (symbol_name))
+    f.write ("  {{")
+
     for i in range(MONO_INST_MAX):
-        f.write ("   0")
-        if i != MONO_INST_MAX - 1:
-            f.write (", ")
-    f.write ("}}, // null entry\n")
-    idx = 1
-    row = 40
+        f.write ("   0")
+        if i != MONO_INST_MAX - 1:
+            f.write (", ")
+    f.write ("}}, // 0 null entry\n")
+    idx = 1
+
+    # Write the heading ever few rows for readability.
+    heading_row = 40
+
+    # Very many of the lines in the table are the same,
```

Comment:
There might be a better way to form this output, in particular an enum, where the enum names are basically the content smushed into an identifier.
That then might be approaching a hand-writable/readable C form.